app/services/TransportEquipmentService.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import models.{SelectableList, TransportEquipment, UserAnswers}
+import pages.sections.external.TransportEquipmentsSection
+
+import javax.inject.Inject
+
+class TransportEquipmentService @Inject() () {
+
+  def getTransportEquipments(userAnswers: UserAnswers): SelectableList[TransportEquipment] = {
+    val equipmentCount = userAnswers.get(TransportEquipmentsSection).get.value.length
+    val transportEquipments = (1 to equipmentCount).map(
+      x => TransportEquipment(x)
+    )
```

Comment:
Or you could just do
`val transportEquipments = userAnswers.get(TransportEquipmentsSection).map(_.value).getOrElse(Nil).zipWithIndex.map(_._2 + 1)`